.github/workflows/ci-core.yml

```diff
@@ -108,69 +91,85 @@ jobs:
       matrix:
         type:
           - cmd: go_core_tests
-            id: core_unit
+            os: ubuntu22.04-32cores-128GB
+            printResults: true
+          - cmd: go_core_tests_integration
             os: ubuntu22.04-32cores-128GB
             printResults: true
           - cmd: go_core_ccip_deployment_tests
-            id: core_unit
             os: ubuntu22.04-32cores-128GB
             printResults: true
           - cmd: go_core_race_tests
-            id: core_race
-            # use 64cores for overnight runs only due to massive number of runs from PRs
-            os: ${{ github.event_name == 'schedule' && 'ubuntu-latest-64cores-256GB' || 'ubuntu-latest-32cores-128GB' }}
+            # use 64cores for certain scheduled runs only
+            os: ${{ needs.run-frequency.outputs.two-per-day-frequency == 'true' && 'ubuntu-latest-64cores-256GB' || 'ubuntu-latest-32cores-128GB' }}
           - cmd: go_core_fuzz
-            id: core_fuzz
             os: ubuntu22.04-8cores-32GB
     name: Core Tests (${{ matrix.type.cmd }})
     # We don't directly merge dependabot PRs, so let's not waste the resources
-    if: github.actor != 'dependabot[bot]'
-    needs: [filter]
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    needs: [filter, run-frequency]
     runs-on: ${{ matrix.type.os }}
     permissions:
       id-token: write
       contents: read
     steps:
       - name: Checkout the repo
         uses: actions/checkout@v4.2.1
-      - name: Setup node
-        if: ${{ needs.filter.outputs.changes == 'true' }}
-        uses: actions/setup-node@v4.0.4
+
+      - name: Change Modtime of Files (cache optimization)
+        shell: bash
+        run: |
+          find . -type f,d -exec touch -r {} -d '1970-01-01T00:00:01' {} \; || true
+
       - name: Setup NodeJS
         if: ${{ needs.filter.outputs.changes == 'true' }}
         uses: ./.github/actions/setup-nodejs
         with:
           prod: "true"
+
       - name: Setup Go
         if: ${{ needs.filter.outputs.changes == 'true' }}
         uses: ./.github/actions/setup-go
+        with:
+          # race/fuzz tests don't benefit from build caching
+          only-modules: ${{ matrix.type.cmd == 'go_core_race_tests' || matrix.type.cmd == 'go_core_fuzz' }}
```

Comment:
Im surprised they dont use the build cache at all

Reply:
They see a benefit when building the chainlink.test binary but that is all. Most of the benefit is from the others, and if we can reduce the number of caches we have, it will lead to longer eviction times which should have an overall benefit for PRs.